shared-libs/transitions/src/transitions/user_replace.js

```diff
@@ -88,71 +65,22 @@ const createNewUser = ({ roles }, { _id, name, phone, parent }) => {
     });
 };
 
-const getReportIdsToReparent = (contactId, timestamp, docsReparented) => {
-  const filters = {
-    date: { from: timestamp + 1, },
-    contact: contactId,
-  };
-  const options = {
-    limit: REPARENT_BATCH_SIZE,
-    skip: docsReparented,
-  };
-  return search.execute('reports', filters, options).then(({ docIds }) => docIds);
-};
-
-const getReports = (keys) => db.medic.allDocs({ keys, include_docs: true })
-  .then(({ rows }) => rows.map(({ doc }) => doc));
-
-const reparentReports = (newContactId, reports) => {
-  if(!reports.length) {
-    return;
-  }
-  reports.forEach(report => report.contact._id = newContactId);
-  return db.medic.bulkDocs(reports);
-};
-
-const reparentReportsFromReplacedUser = (replaceUserDoc, docsReparented = 0) => {
-  const {
-    reported_date,
-    fields: { original_contact_uuid, new_contact_uuid }
-  } = replaceUserDoc;
-  return getReportIdsToReparent(original_contact_uuid, reported_date, docsReparented)
-    .then(reportIds => {
-      if(!reportIds.length) {
-        return;
-      }
-      return getReports(reportIds)
-        .then(reports => reparentReports(new_contact_uuid, reports))
-        .then(() => {
-          if (reportIds.length === REPARENT_BATCH_SIZE) {
-            return reparentReportsFromReplacedUser(replaceUserDoc, docsReparented + REPARENT_BATCH_SIZE);
-          }
-        });
-    });
-};
-
-const replaceUser = (replaceUserDoc) => {
-  return Promise.resolve()
-    .then(() => validateReplaceUserDoc(replaceUserDoc))
-    .then(() => {
-      const { original_contact_uuid, new_contact_uuid } = replaceUserDoc.fields;
-      return  Promise
-        .all([
-          getContact(original_contact_uuid),
-          getContact(new_contact_uuid),
-          users.getUserSettings({ contact_id: original_contact_uuid }),
-        ])
-        .then(([originalContact, newContact, originalUserSettings]) => {
-          validateContactParents(originalContact, newContact);
-          return createNewUser(originalUserSettings, newContact)
-            .then(() => reparentReportsFromReplacedUser(replaceUserDoc))
-            .then(() => users.deleteUser(originalUserSettings.name));
-        });
-    });
+const replaceUser = (originalContact) => {
+  const { _id, replaced: { by } } = originalContact;
+  return Promise
+    .all([
+      getNewContact(by),
+      users.getUserSettings({ contact_id: _id }),
+    ])
+    .then(([newContact, originalUserSettings]) => {
+      return createNewUser(originalUserSettings, newContact)
+        .then(() => users.deleteUser(originalUserSettings.name));
+    })
+    .then(() => originalContact.replaced.status = 'COMPLETE');
```

Comment:
I am not sure about the actual reason, but practically speaking the [eslint config for the shared-libs](https://github.com/medic/cht-core/blob/master/shared-libs/.eslintrc) did not set the `ecmaVersion` until the recent archV3 changes were merged in. So I think this code was covered by the [root eslint config](https://github.com/medic/cht-core/blob/80d0491fd6e3cc39d27ba6f7dbeea317bbdfd42d/.eslintrc#L4) which is still set to `"ecmaVersion": 6`.  

Technically, I think we could update this to use async/await now that we have rebased, but it would have to be reverted when we cherry-pick this back to the `3.17` release...

Reply:
Ah yes that makes a lot of sense, let's keep it that way for now since it's consistent with all the other transitions. We could make it a code health issue to make all transitions use async/await